File: benchmarks/correctness/real_world/python/vcf_filter.py
import json

total = 0
pathogenic = []
with open("real_data/clinvar.vcf") as f:
    for line in f:
        if line.startswith("#"):
            continue
        total += 1
        parts = line.strip().split("\t")
        if len(parts) < 8:
            continue
        info = parts[7]
        for field in info.split(";"):
            if field.startswith("CLNSIG="):
                sig = field.split("=", 1)[1]
                if sig == "Pathogenic" or sig.startswith("Pathogenic/") or sig.startswith("Pathogenic|"):
                    pathogenic.append(parts[0])
                break

per_chrom = {}
for c in pathogenic:
    per_chrom[c] = per_chrom.get(c, 0) + 1

print(json.dumps({"total_variants": total, "pathogenic_count": len(pathogenic), "per_chromosome": per_chrom}))

File: benchmarks/correctness/real_world/r/vcf_filter.R
library(jsonlite)
lines <- readLines("real_data/clinvar.vcf")
lines <- lines[!grepl("^#", lines)]
total <- length(lines)
pathogenic_chroms <- c()
for (line in lines) {
  parts <- strsplit(line, "\t")[[1]]
  if (length(parts) >= 8) {
    info <- parts[8]
    fields <- strsplit(info, ";")[[1]]
    for (f in fields) {
      if (startsWith(f, "CLNSIG=")) {
        sig <- sub("CLNSIG=", "", f)
        if (sig == "Pathogenic" || startsWith(sig, "Pathogenic/") || startsWith(sig, "Pathogenic|")) {
          pathogenic_chroms <- c(pathogenic_chroms, parts[1])
        }
        break
      }
    }
  }
}
per_chrom <- as.list(table(pathogenic_chroms))
per_chrom <- lapply(per_chrom, as.integer)
cat(toJSON(list(total_variants = total, pathogenic_count = length(pathogenic_chroms), per_chromosome = per_chrom), auto_unbox = TRUE))
cat("\n")
